bazel/external/wamr.BUILD

```diff
@@ -1,25 +1,35 @@
-licenses(["notice"])  # Apache 2
-
 load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
 
+licenses(["notice"])  # Apache 2
+
 package(default_visibility = ["//visibility:public"])
 
 filegroup(
     name = "srcs",
     srcs = glob(["**"]),
-    visibility = ["//visibility:public"],
 )
 
 cmake(
-    name = "libiwasm",
+    name = "wamr_lib",
     cache_entries = {
+        "LLVM_DIR": "$EXT_BUILD_DEPS/copy_llvm/llvm/lib/cmake/llvm",
         "WAMR_BUILD_INTERP": "1",
         "WAMR_BUILD_JIT": "0",
         "WAMR_BUILD_AOT": "0",
         "WAMR_BUILD_SIMD": "0",
         "WAMR_BUILD_MULTI_MODULE": "1",
         "WAMR_BUILD_LIBC_WASI": "0",
+        "WAMR_BUILD_TAIL_CALL": "1",
     },
+    cmake_options = ["-GNinja"],
```

Comment:
It looks that WAMR build is currently broken with `Ninja`, which complains about duplicate rules for `libvmlib.a`:
```
ninja: error: build.ninja:1012: multiple rules generate libvmlib.a [-w dupbuild=err]
```

I think there are 2 options:

1. Revert to the default build with `make`.

2. Pass `-w dupbuild=warn` to unblock the build. Using `build_args = ["-v", "--", "-w dupbuild=warn"]` should work, but it errors out somewhere else in pipeline. Note that `Ninja` will still complain, but  both targets are build the same way, so it should be fine, although ideally this should be fixed in WAMR:
```
ninja: warning: multiple rules generate libvmlib.a. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
```

cc @leyao-daily @lum1n0us

Reply:
chose 1. and reverted to use `make`.. but I wonder then why the build in Envoy succeeded..?

Reply:
there is a typo in CMakeLists.txt of WAMR, it will be fixed soon.

Reply:
@mathetake Envoy ships with an older version of `ninja` (1.8.2), for which `cmake` doesn't invoke the `recompact` routine that fails, and to which we cannot append `-w dupbuild=warn`.

I'm not sure which version is used in the `ubuntu-latest` images on GitHub Actions, since officially [it doesn't include `ninja`](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md), and [per their issue tracker](https://github.com/actions/virtual-environments/issues/741) it must be explicitly installed using `sudo apt install ninja-build`.

@lum1n0us thanks!

Reply:
Looks like this was fixed in https://github.com/bytecodealliance/wasm-micro-runtime/commit/17a216748574499bd3a5130e7e6a20b84fe76798, @mathetake do you want to update WAMR and switch back to using `ninja`?

Reply:
https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/162/commits/f3d94963fe5e51adf3e7cff562970f221f7bef35 done! Thanks!